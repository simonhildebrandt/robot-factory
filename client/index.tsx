import * as React from 'react'
import ReactDOM from 'react-dom'

function amanda(lastname: string) {
  return lastname
}

function Thing({x}) {
  return <div>Simon {x+1}</div>
}

function install() {
  const el = document.getElementById('main')
  ReactDOM.render(<Thing x={5}/>, el)
}

export {amanda, Thing, install}
