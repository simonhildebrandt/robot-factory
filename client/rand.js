const plus = (x) => x + 1
export { plus }
