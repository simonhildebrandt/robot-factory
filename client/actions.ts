import {LOAD_ROBOTS, LOADED_ROBOTS} from './store'
import axios from 'axios'


export function loadRobots() {

  return function(dispatch) {
    dispatch({type: LOAD_ROBOTS})

    return axios.get('/robots.json')
    .then(response => {
      dispatch({
        type: LOADED_ROBOTS,
        robots: response.data
      })
    })
  }
}
