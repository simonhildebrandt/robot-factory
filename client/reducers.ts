import {initialState, LOAD_ROBOTS, LOADED_ROBOTS} from './store'


export function reducers(state = initialState, action) {
  switch (action.type) {
    case LOAD_ROBOTS:
      return {...state, loading: true}
    case LOADED_ROBOTS:
      return {...state, robots: action.robots, loading: false}
    default:
      return state
  }
}
