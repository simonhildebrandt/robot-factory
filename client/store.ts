import {createStore} from 'redux'
import {loadRobots} from './actions'
import {reducers} from './reducers'


const LOAD_ROBOTS = 'load robots', LOADED_ROBOTS = 'loaded robots'

const initialState = {
  loading: false,
  robots: []
}

const store = createStore(reducers)

export {store, initialState, LOAD_ROBOTS, LOADED_ROBOTS}
