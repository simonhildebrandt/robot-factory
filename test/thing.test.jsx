import {plus} from '../client/rand'
import {amanda, Thing} from '../client/index.tsx'
import renderer from 'react-test-renderer'
import React from 'react'


test('adds 1 to 5 to get 6', () => {
  expect(plus(5)).toBe(6);
});

test('amanda function', () => {
  expect(amanda('x')).toBe('x');
});

test('thing render', () => {
  const component = renderer.create(<Thing x={3}/>)
  let tree = component.toJSON();
  console.log(tree);
  expect(tree).toMatchSnapshot();
});
