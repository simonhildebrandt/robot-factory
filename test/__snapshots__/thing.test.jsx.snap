// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`thing render 1`] = `
<div>
  Simon 
  4
</div>
`;
